bridgetown-core/lib/bridgetown-core/configuration.rb

```diff
@@ -144,9 +144,11 @@ def run_initializers!(context:)
       Bridgetown.logger.debug "Initializing:", "Running initializers with `#{context}' context in:"
       Bridgetown.logger.debug "", initializers_file
       self.init_params = {}
+      cached_url = url&.include?("//localhost") ? url : nil
```

Comment:
**question:** How else other than running `bin/bt dev` can the value be set at this point? I'm unclear why we need to grab this `cached_url` because of my lack of understanding here.

Do we need to grab it because the port is overrideable? Because it can be set in other ways?

My hypothesis is that perhaps a `bridgetown.config.yml` has already been processed and we need to account for a value in there. Am I on the right track?

Reply:
@michaelherold it's overridden by the `start` command when in development environment before any other configuration has been loaded, even if there's a `url` key in the YAML config. But then when the initializers code is executed, setting `url` was overriding the `localhost` overwrite 😅. We want to undo that if necessary. There may be a more elegant way to refactor this in the near future…